syntax = "proto3";

service Authentication {
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc VerifyJWT(VerifyJWTRequest) returns (EmptyResponse);
    rpc Registration(RegistrationRequest) returns (RegistrationResponse);
    rpc ForgotPassword(ForgotPasswordRequest) returns (EmptyResponse);
    rpc ChangePassword(ChangePasswordRequest) returns (EmptyResponse);
}

message EmptyResponse{}

message VerifyJWTRequest{
    string token = 1;
}

message LoginRequest {
    string email = 1;
    string password = 2;
}

message LoginResponse {
    string token = 1;
}

message RegistrationRequest {
    string email = 1;
    string nickname = 2;
    string phone_number = 4;
    string password = 5;
}

message RegistrationResponse {
    string token = 1;
}

message ForgotPasswordRequest {
    string email = 1;
}


message ChangePasswordRequest {
    string email = 1;
    string old_password = 2;
    string new_password = 3;
}

